samples/sample_multi_transcode/src/pipeline_transcode.cpp

```diff
@@ -1271,7 +1271,7 @@ mfxStatus CTranscodingPipeline::Encode()
                 // Getting next frame
                 while (MFX_ERR_MORE_SURFACE == curBuffer->GetSurface(DecExtSurface))
                 {
-                    if (MFX_ERR_NONE != curBuffer->WaitForSurfaceInsertion(MSDK_SURFACE_WAIT_INTERVAL)) {
+                    if (!curBuffer->WaitForSurfaceInsertion(MSDK_SURFACE_WAIT_INTERVAL)) {
```

Comment:
if buffering is not allowed (`m_IsBufferingAllowed == false`) `GetSurface` returns `MFX_ERR_MORE_SURFACE` and we get into `WaitForSurfaceInsertion`. The latter will wait for `m_bInsEventHappened` that becomes true at `AddSurface` when `m_IsBufferingAllowed == true`, so we'll exit w/ `MFX_ERR_NOT_FOUND` won't 'we ?

Reply:
That should not happen, because ```m_IsBufferingAllowed``` is set to ```false``` only in ```CancelBuffering``` which happens only after the main loop of encoding is finished (i.e. no more encoding will occur). So buffering is canceled right before application shutdown. So current code (as well as previous implementation) should work